src/test/java/com/auth0/client/MockServer.java

```diff
@@ -107,6 +107,15 @@ public void jsonResponse(String path, int statusCode) throws IOException {
                 .setBody(readTextFile(path));
         server.enqueue(response);
     }
+    
+    public void rateLimitReachedResponse(long limit, long remaining, long reset) throws IOException {
+        MockResponse response = new MockResponse()
+                .setResponseCode(429)
+                .addHeader("X-RateLimit-Limit", String.valueOf(limit))
```

Comment:
You might want to add a check whether the arg is "-1" to avoid setting the header, so you can test that.

Reply:
All suggestions have been done!